### PR TITLE
Spoken digit

### PR DESCRIPTION
**Pull request by jorshi**
Add TFDS Spoken Digit. This also serves as a kfold preprocessing pipeline for CI.

### CODE REVIEW

File: hearpreprocess/spoken_digit.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+generic_task_config = {
+    "task_name": "spoken_digit",
+    "version": "1.0.9",
+    "embedding_type": "scene",
+    "prediction_type": "multiclass",
+    "split_mode": "new_split_kfold",
+    "nfolds": 5,
+    # mean: 0.41s, 75th: 0.47s, 90th: 0.56s, max: 2.28s
+    "sample_duration": 1.0,
```
Comment (turian) [inline hearpreprocess/spoken_digit.py:20]: What is the 95th percentile? Maybe we shouldn't be extending this past 95th percentile and that is important?

File: hearpreprocess/spoken_digit.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    Receives a task_config dictionary, downloads and extracts the correct files from
+    TFDS and prepares the ExtractMetadata task class for the train split
+    """
+    if task_config["mode"] == "small":
+        # Small mode uses a sampled version of TF dataset - downloads directly
+        # from a URL as opposed to using the tfds methods.
+        download_tasks = pipeline.get_download_and_extract_tasks(task_config)
+    else:
+        download_tasks = tfds_pipeline.get_download_and_extract_tasks_tfds(task_config)
+
```
Comment (khumairraj) [inline hearpreprocess/spoken_digit.py:69]: This is good

**Review by turian — APPROVED** [review]

File: hearpreprocess/spoken_digit.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    "embedding_type": "scene",
+    "prediction_type": "multiclass",
+    "split_mode": "new_split_kfold",
+    "nfolds": 5,
+    # mean: 0.41s, 75th: 0.47s, 90th: 0.56s, max: 2.28s
+    "sample_duration": 1.0,
+    "evaluation": ["top1_acc"],
+    "tfds_task_name": "spoken_digit",
+    "tfds_task_version": "1.0.9",
+    "extract_splits": ["train"],
```
[the remark this answers is not in the document]
Comment (jorshi) [reply hearpreprocess/spoken_digit.py:24]: Having a hard time finding this dataset in the literature. The test/train splits come from here: https://github.com/Jakobovski/free-spoken-digit-dataset

File: hearpreprocess/spoken_digit.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+from typing import Any, Dict
+
+import luigi
+
+import hearpreprocess.pipeline as pipeline
+import hearpreprocess.tfds_pipeline as tfds_pipeline
+
+generic_task_config = {
+    "task_name": "spoken_digit",
+    "version": "1.0.9",
```
[the remark this answers is not in the document]
Comment (jorshi) [reply hearpreprocess/spoken_digit.py:14]: The github has 1.0.9 and 1.0.10

**Pull request merged**